matcher.go

```diff
@@ -208,3 +209,21 @@ func AllNumberGroupsAreExactlyPresent(
 		strings.HasSuffix(candidateGroups[candidateNumberGroupIndex],
 			formattedNumberGroups[0]))
 }
+
+// Returns whether the given national number (a string containing only decimal digits) matches
+// the national number pattern defined in the given PhoneNumberDesc message.
+func MatchNationalNumber(number string, numberDesc PhoneNumberDesc, allowPrefixMatch bool) bool {
+	nationalNumberPattern := numberDesc.GetNationalNumberPattern()
+	// We don't want to consider it a prefix match when matching non-empty input against an empty pattern.
+	if len(nationalNumberPattern) == 0 {
+		return false
+	}
+	patP := `^(?:` + nationalNumberPattern + `)$` // Strictly match
+	regex := regexFor(patP)
+	return match(number, regex, allowPrefixMatch)
+}
+
+// TODO: review this matches java version?
+func match(number string, pattern *regexp.Regexp, allowPrefixMatch bool) bool {
+	return pattern.MatchString(number)
```

Comment:
don't we need to support `allowPrefixMatch` ?

Reply:
updated with above comment